src/Whim/Template/whim.config.csx

```diff
@@ -113,7 +113,8 @@ void DoConfig(IContext context)
 			(id) => SliceLayouts.CreatePrimaryStackLayout(context, sliceLayoutPlugin, id),
 			(id) => SliceLayouts.CreateSecondaryPrimaryLayout(context, sliceLayoutPlugin, id),
 			(id) => new FocusLayoutEngine(id),
-			(id) => new TreeLayoutEngine(context, treeLayoutPlugin, id)
+			(id) => new TreeLayoutEngine(context, treeLayoutPlugin, id),
+			(id) => new FreeLayoutEngine(context, id)
 		};
 }
 
```

Comment:
> There's a separate issue with the GapsPlugin, where that will try to add gaps, but that should be fairly simple to fix - the GapsPlugin can avoid adding gaps when it finds that the leaf engine is the new floating engine.

I tried to do something like this in the `DoLayout` of GapsLayoutEngine, but all the windows seems flickering.

```
		foreach (IWindowState windowState in InnerLayoutEngine.DoLayout(proxiedRect, monitor))
		{
			if (InnerLayoutEngine.GetLayoutEngine<FreeLayoutEngine>() is not null)
			{
				yield return new WindowState
				{
					Window = windowState.Window,
					Rectangle = windowState.Rectangle,
					WindowSize = windowState.WindowSize
				};
			}
```

Reply:
I think the issue is that you're passing in the `proxiedRect` to `InnerLayoutEngine.DoLayout`. If you detect a `FreeLayoutEngine`, you can pass in the original `rectangle` to `DoLayout`.

Reply:
I think the issue is that you're passing in the `proxiedRect` to `InnerLayoutEngine.DoLayout`. If you detect a `FreeLayoutEngine`, you can pass in the original `rectangle` to `DoLayout`.

Reply:
It is not enough.
I would have expected that the `GetLayoutEngine` (as it is recursive) would always return the `FreeLayoutEngine` in case we are using the it.
However, it does not seem to be the case, because it does not always enter in the `if`.

Reply:
I might be wrong, but I think adding the following at the start of `GapsLayoutEngine.DoLayout` should do the trick:

```csharp
		if (InnerLayoutEngine.GetLayoutEngine<FreeLayoutEngine>() is not null)
		{
			foreach (IWindowState windowState in InnerLayoutEngine.DoLayout(rectangle, monitor))
			{
				yield return windowState;
			}

			yield break;
		}
```

The `yield break` specifies that the iterator has come to an end.

Reply:
I might be wrong, but I think adding the following at the start of `GapsLayoutEngine.DoLayout` should do the trick:

```csharp
		if (InnerLayoutEngine.GetLayoutEngine<FreeLayoutEngine>() is not null)
		{
			foreach (IWindowState windowState in InnerLayoutEngine.DoLayout(rectangle, monitor))
			{
				yield return windowState;
			}

			yield break;
		}
```

The `yield break` specifies that the iterator has come to an end.

Reply:
Yes, I agree, it is a better solution.

Reply:
Done